content/en/data_observability/datasets.md

```diff
@@ -0,0 +1,197 @@
+    while (schemaResultSet.next()) {
+        numberOfSchemasGranted += 1;
+        var schemaAndRoleSuffix = ' in schema "' + DATABASENAME + '"."' + 
+        schemaResultSet.getColumnValue('SCHEMA_NAME') + '" to role ' + ROLENAME + ';'
+
+        snowflake.execute({ sqlText: 'grant USAGE on schema "' + DATABASENAME + '"."' +  
+        schemaResultSet.getColumnValue('SCHEMA_NAME') + '" to role ' + ROLENAME + ';'});
+        snowflake.execute({ sqlText: 'grant SELECT on all tables' + schemaAndRoleSuffix});
+        snowflake.execute({ sqlText: 'grant SELECT on all views' + schemaAndRoleSuffix});
+        snowflake.execute({ sqlText: 'grant SELECT on all event tables' + schemaAndRoleSuffix});
+        snowflake.execute({ sqlText: 'grant SELECT on all external tables' + schemaAndRoleSuffix});
+        snowflake.execute({ sqlText: 'grant SELECT on all dynamic tables' + schemaAndRoleSuffix});
+        snowflake.execute({ sqlText: 'grant SELECT on future tables' + schemaAndRoleSuffix});
+        snowflake.execute({ sqlText: 'grant SELECT on future views' + schemaAndRoleSuffix});
+        snowflake.execute({ sqlText: 'grant SELECT on future event tables' + schemaAndRoleSuffix});
+        snowflake.execute({ sqlText: 'grant SELECT on future external tables' + schemaAndRoleSuffix});
+        snowflake.execute({ sqlText: 'grant SELECT on future dynamic tables' + schemaAndRoleSuffix});
+    }
+    
+    return 'Granted access to ' + numberOfSchemasGranted + ' schemas';
+    $$
+    ;
+
+    GRANT USAGE ON DATABASE IDENTIFIER($database_name) TO ROLE IDENTIFIER($role_name);
+    CALL grantFutureAccess('<DATABASE_NAME>', '<ROLE_NAME>');
+    ```
+
+1. (Optional) If your organization uses [Snowflake event tables][2], you can grant the Datadog role access to them.
+
+    ```sql
+    -- Grant usage on the database, schema, and table of the event table
+    GRANT USAGE ON DATABASE <EVENT_TABLE_DATABASE> TO ROLE IDENTIFIER($role_name);
+    GRANT USAGE ON SCHEMA <EVENT_TABLE_DATABASE>.<EVENT_TABLE_SCHEMA> TO ROLE IDENTIFIER($role_name);
+    GRANT SELECT ON TABLE <EVENT_TABLE_DATABASE>.<EVENT_TABLE_SCHEMA>.<EVENT_TABLE_NAME> TO ROLE IDENTIFIER($role_name);
+
+    -- Snowflake-provided application roles for event logs
+    GRANT APPLICATION ROLE SNOWFLAKE.EVENTS_VIEWER TO ROLE IDENTIFIER($role_name);
+    GRANT APPLICATION ROLE SNOWFLAKE.EVENTS_ADMIN TO ROLE IDENTIFIER($role_name);
+
+    ```
+
+After completing the Snowflake setup, configure the Snowflake integration in Datadog.
+
+1. On the [Snowflake integration tile][3], click **Add Snowflake account**.
+1. Enter your Snowflake account URL.
+1. Under **Logs**, turn on:
+   - **Query History Logs**
+   - **Enable Query Logs with Access History**
+1. Under **Data Observability**, turn on:
+   - **Enable Data Observability for Snowflake tables**
+1. Set the **User Name** to `DATADOG_USER`.
+1. Under **Configure a key pair authentication**, upload your unencrypted RSA private key.
```

Comment:
```suggestion
1. On the **General** tab:
   - Enter your Snowflake **Account URL**.
   - Set the **Snowflake User Name** to `DATADOG_USER`.
1. On the **Resource Collection** tab, under **Logs**, turn on:
   - **Query History Logs**
   - **Enable Query Logs with Access History**
1. Under **Data Observability**, turn on:
   - **Enable Data Observability for Snowflake tables**
1. Under **Configure a key pair authentication**, upload your unencrypted RSA private key.
1. Click **Save**.
```
This is all very picky; I was comparing it with what I was seeing in the UI and getting a bit confused. If I'm looking at an outdated version, ignore me!